src/keylessBackup/SignInWithEmail.tsx

```diff
@@ -120,48 +125,58 @@ function SignInWithEmail({ route }: Props) {
     bottomSheetRef.current?.snapToIndex(0)
   }
 
-  const onPressGoogle = async () => {
-    setLoading(true)
+  const setLoading = (provider: OAuthProvider, value: boolean) => {
+    if (provider === 'google-oauth2') {
+      setLoadingGoogle(value)
+    } else {
+      setLoadingApple(value)
+    }
+  }
+
+  const onPressSignIn = async (signInWith: OAuthProvider) => {
```

Comment:
```suggestion
  const onPressSignIn = async (oAuthProvider: OAuthProvider) => {
```
or just `provider` to make it more obvious